### PR TITLE
CEO#4.6 - Rename all Leader references to CEO

### PR DESCRIPTION
**Pull request by d-little**
_PR#6 in a set of card+deck related PRs.  PR-thread 4.x now includes .vue files_

### Rename all Leader references to CEO

This began with some basic .vue changes, but I wanted to start naming things CEO... it cascaded and everything is now CEO.
Sorry... but there's only a small handful of _actual_ changes so it should be OK.

### SelectInitialCards.ts

Pretty much identical to the existing Prelude selector.

### misc.vue additions

These are basic changes:
src/client/components/OtherPlayer.vue
src/client/components/create/CorporationsFilter.vue
src/client/components/create/CreateGameForm.vue

### Rename Leader to CEO

Everything else is just the rename from Leader to CEO.  All references to Leader remaining should be Turmoil, everything else is CEO.

### CODE REVIEW

File: src/server/GameCards.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export class GameCards {
       [gameOptions.communityCardsOption, COMMUNITY_CARD_MANIFEST],
       [gameOptions.moonExpansion, MOON_CARD_MANIFEST],
       [gameOptions.pathfindersExpansion, PATHFINDERS_CARD_MANIFEST],
-      [gameOptions.leadersExtension, LEADER_CARD_MANIFEST],
+      [gameOptions.ceoExtension, CEO_CARD_MANIFEST], // TODO: Rename to CEO_CARD_MANIFEST
```
Comment (kberg) [inline src/server/GameCards.ts:58]: Why is there a TODO?

File: src/server/GameOptions.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ export const DEFAULT_GAME_OPTIONS: GameOptions = {
   customColoniesList: [],
   customCorporationsList: [],
   customPreludes: [],
-  customLeaders: [],
+  customCeos: [],
```
Comment (kberg) [inline src/server/GameOptions.ts:76]: When you get a chance, keep these in alphabetical order.

File: src/server/GameOptions.ts [hunk 3]
```diff
@@ -86,7 +86,7 @@ export const DEFAULT_GAME_OPTIONS: GameOptions = {
   moonExpansion: false,
   moonStandardProjectVariant: false,
   pathfindersExpansion: false,
-  leadersExtension: false,
+  ceoExtension: false,
```
Comment (kberg) [inline src/server/GameOptions.ts:89]: Alphabetize

File: src/server/cards/AllCards.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ export const ALL_MODULE_MANIFESTS: Array<ModuleManifest> = [
   VENUS_CARD_MANIFEST,
   COLONIES_CARD_MANIFEST,
   PRELUDE_CARD_MANIFEST,
-  LEADER_CARD_MANIFEST,
+  CEO_CARD_MANIFEST,
```
Comment (kberg) [inline src/server/cards/AllCards.ts:24]: These manifests are in order with official ones first, so let's keep it that way. In fact, put this one after Pathfinders, which matches other orderings.

File: src/server/models/ServerModel.ts [hunk 5]
```diff
@@ -114,13 +114,15 @@ export class Server {
...
       dealtPreludeCards: this.getCards(player, player.dealtPreludeCards),
+      dealtCeoCards: this.getCards(player, player.dealtCeoCards),
       dealtProjectCards: this.getCards(player, player.dealtProjectCards),
       draftedCorporations: this.getCards(player, player.draftedCorporations),
       draftedCards: this.getCards(player, player.draftedCards, {showCalculatedCost: true}),
       game: this.getGameModel(player.game),
       id: player.id,
       pickedCorporationCard: player.pickedCorporationCard ? this.getCards(player, [player.pickedCorporationCard]) : [],
       preludeCardsInHand: this.getCards(player, player.preludeCardsInHand),
+      ceoCardsInHand: this.getCards(player, player.ceoCardsInHand),
```
Comment (kberg) [inline src/server/models/ServerModel.ts:125]: Keep alphabetized.

**Review by kberg — COMMENTED** [review]
Just some cleanups, this is great.

File: src/client/components/SelectInitialCards.vue [hunk 6]
```diff
@@ -277,7 +303,15 @@ export default (Vue as WithRefs<Refs>).extend({
       return this.playerView.dealtCorporationCards.some((card) => card.name === CardName.ARIDOR);
     },
     hasPrelude() {
-      return this.playerinput.options?.length === 3;
+      // Was previously this:
```
[the remark this answers is not in the document]
Comment (d-little) [reply src/client/components/SelectInitialCards.vue:306]: #5285-1

File: src/client/components/SelectInitialCards.vue [hunk 7]
```diff
@@ -294,9 +328,26 @@ export default (Vue as WithRefs<Refs>).extend({
       }
       return option;
     },
+    ceoCardOption() {
+      let counter = 1;
```
[the remark this answers is not in the document]
Comment (d-little) [reply src/client/components/SelectInitialCards.vue:332]: #5285-2

File: src/server/GameCards.ts (hunk 1 above)
Comment (d-little) [reply src/server/GameCards.ts:58]: #5285-3

File: src/server/GameOptions.ts (hunk 2 above)
Comment (d-little) [reply src/server/GameOptions.ts:76]: #5285-4

File: src/server/GameOptions.ts (hunk 3 above)
Comment (d-little) [reply src/server/GameOptions.ts:89]: #5285-4

File: src/server/cards/AllCards.ts (hunk 4 above)
Comment (d-little) [reply src/server/cards/AllCards.ts:24]: #5285-5

File: src/server/models/ServerModel.ts (hunk 5 above)
Comment (d-little) [reply src/server/models/ServerModel.ts:125]: #5285-6

**Comment by d-little** [comment]
I pulled out the SIC.vue changes as requested, should be easier to merge now.  It still has a handful of other vue additions, and the CEO renames, I can pull out all the .vue changes too if necessary

**Review by kberg — APPROVED** [review]

**Pull request merged**